Upgrade FuzzUtil to support Bellatrix

jtraglia opened the pull request:
## PR Description

This PR updates the fuzz utility (`FuzzUtil`) from Phase0 to Bellatrix. I thought about creating a new `FuzzUtilBellatrix` class but decided against it because I do not think we need to support old versions (Phase0 & Altair).

I am not sure how the original developers of this created all of the [pyspec](https://github.com/ethereum/consensus-specs/tree/dev/tests/core/pyspec) yaml files. I do not see any [generator](https://github.com/ethereum/consensus-specs/tree/dev/tests/generators) options for this. I will need to either figure out how to do this or delete all of the old yaml files to avoid any confusion.

Fixes #5216.

## Documentation

- [x] I thought about documentation and added the `doc-change-required` label to this PR if updates are required.

## Changelog

- [x] I thought about adding a changelog entry, and added one if I deemed necessary.

jtraglia committed:
Start to add FastUtil Bellatrix support

Not entirely sure I'm doing things properly, but I'll figure it out soon. This
commit adds a fuzzSyncAggregate method (Altair) with no testing. Currently
several FuzzUtilTests are failing; I'll fix those later.

jtraglia committed:
Add fuzzExecutionPayload method

jtraglia committed:
Add *FuzzInputTest classes for the two new ones

jtraglia committed:
Update variables in FuzzUtilTest to Bellatrix

Don't worry, the tests still fail. I'll get around to those eventually.

jtraglia committed:
Fix FuzzUtilTests

So all of the tests pass now! This involved replacing all of the ssz files with
their respective bellatrix versions. I needed to run `gen_operations` and
`gen_sanity` in ethereum/consensus-specs to make them.

I needed to modify the `dump_ssz_fn` in gen_runner.py to not snappy compress
the serialized data before writing it to disk. Not sure if this is new or what,
but the test suite does not work with snappy data. To be fair, that's
unnecessary work for fuzzing.

Next, I need to update the yaml files next. I think I'm going to need to modify
the generation functions to tell it to output yaml. Not sure how the original
developers of this did that.

jtraglia committed:
Add tests for the two new operations

jtraglia committed:
Clean up fuzzing harness comment

Sorry this was bothering me. It wasn't properly justified. I also added a bit
about where the test case files came from.

jtraglia committed:
Remove yaml files

jtraglia committed:
Replace ssz files with ssz_snappy files

jtraglia committed:
Remove two extra yaml files

Oops these were added by accident.

jtraglia committed:
Merge branch 'master' into bellatrix-fuzz

ajsutton commented on fuzz/src/main/java/tech/pegasys/teku/fuzz/input/SyncAggregateFuzzInput.java:
```diff
@@ -0,0 +1,61 @@
+    extends Container2<SyncAggregateFuzzInput, BeaconState, SyncAggregate> {
+
+  public static ContainerSchema2<SyncAggregateFuzzInput, BeaconState, SyncAggregate> createSchema(
+      final SpecVersion spec, final SpecConfigAltair config) {
```
nit: Can remove the `config` parameter from here (and then probably also from the `SyncAggregateFuzzInput` constructor).

jtraglia replied:
Good catch. I needed it originally but not anymore. It's been removed.

ajsutton reviewed (commented):
I can't get GitHub to comment on it but the new `sync_aggregate.ssz_snappy` file seems to have come through as a text file but needs to be marked as binary in git.  Probably should add `*.ssz_snappy binary` to `.gitattributes` and may need to remove then re-add this file to get it to apply.  Autodetection magic seems to have worked fine for the other files.

Otherwise this looks good. Thanks.

jtraglia committed:
Remove unnecessary specConfig from FuzzUtil

jtraglia committed:
Mark ssz_snappy files as binary in gitattributes

jtraglia committed:
Delete sync_aggr ssz_snappy file (will add back)

This is a little jank, but we need to delete and re-add this file back so that
git treats it as a binary file. I do not think there is another way to do this.

jtraglia committed:
Add back sync_aggregate.ssz_snappy

jtraglia committed:
Set !eol for binary files in gitattributes

jtraglia committed:
Remove sync_aggregate.ssz_snappy (will add back)

jtraglia committed:
Add back sync_aggregate.ssz_snappy

ajsutton committed:
Merge branch 'master' into bellatrix-fuzz

ajsutton committed:
Don't declare all files as binary

ajsutton committed:
Merge branch 'master' into bellatrix-fuzz
